### PR TITLE
Ftr: uniform router support

### PR DESCRIPTION
**Pull request by LaurenceLiZhixin**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
This pr add uniform router feature, and delete old routers' impl.
User can use local config file, or load config and refresh router by using  k8s CRD. 

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by AlexStocks** [comment]
If this pr has not been completed, pls add [WIP] before ur pr title.

File: cluster/router/chain/chain.go [hunk 1]
```diff
@@ -234,10 +152,14 @@ func NewRouterChain(url *common.URL) (*RouterChain, error) {
 	}
 
 	routers := make([]router.PriorityRouter, 0, len(routerFactories))
+
 	for key, routerFactory := range routerFactories {
-		r, err := routerFactory().NewPriorityRouter(url, chain.notify)
+		if virtualServiceConfigByte == nil || destinationRuleConfigByte == nil {
+			logger.Errorf("virtual Service Config or destinationRule Confi Byte may be empty, pls check your CONF_VIRTUAL_SERVICE_FILE_PATH and CONF_DEST_RULE_FILE_PATH env is correctly point to your yaml file\n")
```
Comment (wenxuwan) [inline cluster/router/chain/chain.go:158]: change the log level to warn

File: cluster/router/uniform/factory_test.go [hunk 2]
```diff
@@ -15,26 +15,18 @@
...
-package tag
-
-import (
-	"testing"
-)
+package uniform
 
 import (
 	"github.com/stretchr/testify/assert"
+	"testing"
```
Comment (wenxuwan) [inline cluster/router/uniform/factory_test.go:22]: move  package to first import()

Comment (wenxuwan) [inline cluster/router/uniform/factory_test.go:22]: move  package to first import()

**Review by AlexStocks — APPROVED** [review]

**Review by ztelur — APPROVED** [review]

**Review by beiwei30 — APPROVED** [review]
LGTM.

**Pull request merged**